frontend/src/components/DownloadAnalysisLink/DownloadAnalysisLink.tsx

```diff
@@ -24,7 +25,13 @@ DownloadAnalysisLink.displayName = 'DownloadAnalysisLink'
 
 function onClick(store: Store) {
   let method = store.get('method')
+  const selectedDataset = store.get('selectedDataset')
   return () => {
+    ReactGA.event({
+      category: 'Analysis',
+      action: 'Download analysis results',
+      label: selectedDataset ? selectedDataset.name : 'Unknown Dataset'
```

Comment:
Maybe use a ! instead? Can we arrive here with an empty dataset?

Reply:
the dataset is nullable so it won't compile otherwise.